<launch>

  <arg name="arg_record_mode" default="2" />
  <arg name="arg_ev_mode" default="1" />

  <node pkg="wind_turbine_insp" type="vision_node" name="vision_node" required="true" output="screen" >
   
    <remap from="/wind_turbine_insp/gimbal_image"
           to="/typhoon_h480/gimbal/camera/image" />

    <param name="photo_intr_fx"       value="1461.44326"       type="double" /> 
    <param name="photo_intr_fy"       value="1461.44326"       type="double" /> 
    <param name="photo_intr_cx"       value="0.5"       type="double" /> 
    <param name="photo_intr_cy"       value="0.5"       type="double" /> 

    <param name="video_intr_fx"       value="1461.44326"       type="double" /> 
    <param name="video_intr_fy"       value="1461.44326"       type="double" /> 
    <param name="video_intr_cx"       value="0.5"       type="double" /> 
    <param name="video_intr_cy"       value="0.5"       type="double" /> 

    <param name="rate_image_proc"       value="2.0"         type="double" />
    <param name="rate_adjust_gim"       value="2.0"         type="double" />
    <param name="rate_set_camera_ev"    value="0.2"         type="double" />
    <param name="rate_shoot_interval"   value="0.5"         type="double" />

    <param name="video_record_mode"    value="$(arg arg_record_mode)"  type="int" />
    <param name="ev_mode"              value="$(arg arg_ev_mode)"  type="int" />


    <param name="area_radius_ori"          value="80"          type="int" />

    <param name="area_mean_max"        value="180.0"         type="double" />
    <param name="area_mean_min"        value="120.0"         type="double" />
    <param name="mean_offset"             value="10"           type="int" />
    <param name="adjust_k"             value="10"           type="int" />

    <param name="set_adj_limit"         value="2"          type="int" />

    <param name="msg_text_scale"       value="1.6"           type="double" />
    <param name="msg_text_thick"       value="8"            type="int" />

    <param name="enable_line"            value="false"           type="bool" />

    <param name="canny_min"            value="40"           type="double" />
    <param name="canny_max"            value="120"           type="double" />
    <param name="hough_thold"          value="30"            type="int" />
    <param name="min_line_len"         value="60"           type="double" />
    <param name="max_line_gap"         value="50"           type="double" />
    <param name="max_line_num"         value="10"            type="int" />

    <param name="line_detect_err"       value="10.0"            type="double" />

    <param name="compensate_k"       value="0.0037037"            type="double" />

    <param name="focus_len_ori"        value="1461.44326"    type="double" />

  </node>

</launch>